spec/lib/miq_expression_spec.rb

```diff
@@ -3254,4 +3254,35 @@
       expect(actual).to eq([["My Company Tags : Environment", "managed-env"]])
     end
   end
+
+  describe "miq_adv_search_lists" do
+    it ":exp_available_fields" do
+      options = {:include_model                              => true,
+                 :typ                                        => 'field',
+                 :disallow_loading_virtual_custom_attributes => false}
+      expect(described_class).to receive(:model_details).with(Vm, options)
```

Comment:
These tests are tightly coupled to the implementation by mocking the system under test. I'd have more confidence if they instead verified against what was returned, treating this method as more of a black box.